### ISSUE TITLE
Vocabulary size for sentencepiece

### ISSUE THREAD

**Issue by dust0x** [n-waves/ulmfit-multilingual#14]
Before commit [`5ffbe8b`](https://github.com/n-waves/ulmfit-multilingual/commit/5ffbe8ba5ce6146188a64e4427fa42d781921920) (which was required for making sure the vocab for testing would be small) the sentencepiece model's vocabulary size defaulted to 30,000 whereas, now it'll be the same as `pretrain_lm()`'s default of 60,000.

In my experience in training the sentencepiece, I've found that very large vocabularies degrade performance and in most cases lead to memory failure, at least on small/medium machines (it did on mine). Even the [official documentation](https://github.com/google/sentencepiece#train-sentencepiece-model) suggests vocabularies of up to 32,000 tokens.

Any thoughts on how we may resolve this discrepancy?

**Comment by sebastianruder** [comment]
Agreed. Maybe we can just add a line that sets the value to `30,000` if the default of `60,000` is used for sentence piece?

**Comment by dust0x** [comment]
Yes, just a check will do but my concern was that how doing that would take away from a user.
